cpp/src/io/text/multibyte_split.cu

```diff
@@ -275,180 +517,126 @@ int64_t multibyte_split_scan_full_source(cudf::io::text::data_chunk_source const
-                                              std::string const& delimiter,
-                                              byte_range_info byte_range,
-                                              rmm::cuda_stream_view stream,
-                                              rmm::mr::device_memory_resource* mr,
-                                              rmm::cuda_stream_pool& stream_pool)
-{
-  CUDF_FUNC_RANGE();
-  auto const trie = cudf::io::text::detail::trie::create({delimiter}, stream);
-
-  CUDF_EXPECTS(trie.max_duplicate_tokens() < multistate::max_segment_count,
-               "delimiter contains too many duplicate tokens to produce a deterministic result.");
-
-  CUDF_EXPECTS(trie.size() < multistate::max_segment_value,
-               "delimiter contains too many total tokens to produce a deterministic result.");
-
-  // Empty byte ranges could still produce output if the beginning overlapped with the beginning of
-  // a field with the logic below. Best disallow them instead.
-  CUDF_EXPECTS(byte_range.size() > 0, "byte range cannot be empty");
-
-  auto concurrency = 2;
-  // must be at least 32 when using warp-reduce on partials
-  // must be at least 1 more than max possible concurrent tiles
-  // best when at least 32 more than max possible concurrent tiles, due to rolling `invalid`s
-  auto num_tile_states  = std::max(32, TILES_PER_CHUNK * concurrency + 32);
-  auto tile_multistates = scan_tile_state<multistate>(num_tile_states, stream);
-  auto tile_offsets     = scan_tile_state<int64_t>(num_tile_states, stream);
-
-  auto streams = get_streams(concurrency, stream_pool);
-
-  auto bytes_total =
-    multibyte_split_scan_full_source(source,
-                                     trie,
-                                     tile_multistates,
-                                     tile_offsets,
-                                     cudf::device_span<int64_t>(static_cast<int64_t*>(nullptr), 0),
-                                     stream,
-                                     streams);
-
-  // allocate results
-  auto num_tiles =
-    cudf::util::div_rounding_up_safe(bytes_total, static_cast<int64_t>(ITEMS_PER_TILE));
-  auto num_results = tile_offsets.get_inclusive_prefix(num_tiles - 1, stream);
-
-  auto string_offsets = rmm::device_uvector<int64_t>(num_results + 2, stream);
-
-  // first and last element are set manually to zero and size of input, respectively.
-  // kernel is only responsible for determining delimiter offsets
-  string_offsets.set_element_to_zero_async(0, stream);
-  string_offsets.set_element_async(string_offsets.size() - 1, bytes_total, stream);
-
-  // kernel needs to find first and last relevant offset., as well as count of relevant offsets.
+  // if the input was empty, we didn't find a delimiter at all,
+  // or the first delimiter was also the last: empty output
+  if (chunk_offset == 0 || !first_offset.has_value() || first_offset == last_offset) {
```

Comment:
are all of these cases tested?

Reply:
Empty inputs yes, I added a test for `!first_noffset.has_value()`, `first_offset == last_offset` should be tested implicitly by `SmallInputAllPossibleRanges`